### PR TITLE
Add ocamlformat and update CI pipeline

### PR DESCRIPTION
**Pull request by shonfeder**
- Add an `.ocamlformat` config
- Reformat code base
- CI pipeline updates
  - Use github actions v2
  - Run formatting check
  - Add missing compiler versions
  - Enable dune-cache

I'll wait to merge this until #232 (and other substantial outstanding PRs?) are merged in.

In the meantime, @sonologico, whenever you have a chance, let me know if you are OK with these formatting, or if you have strong feelings by a different config.

### CODE REVIEW

File: bin/main.ml [hunk 1]
```diff
@@ -1,14 +1,22 @@
 let with_open_in fn f =
   let ic = open_in fn in
   match f ic with
-  | r -> close_in ic; r
-  | exception e -> close_in_noerr ic; raise e
+  | r           ->
+      close_in ic;
+      r
+  | exception e ->
```
Comment (sonologico) [inline bin/main.ml:7]: Everything looks good to me, except for the aligned arrows. I don't feel strongly about it, so if you really prefer it that way, I'm ok with it.

Having said that, the rule is either not applied consistently or I don't understand when it should apply.

Comment (shonfeder) [reply bin/main.ml]: I am ok either way. I agree it's awkward in these cases, but I tend to feel like it pays off when you have lots of similarly sized matches.

I'm not sure what's up with the inconsistency!

I've removed that rule. Let me know if you like this better :)

Comment (sonologico) [reply bin/main.ml]: I prefer it this way, but I'm ok with the other way if we figure out why it's inconsistent.

**Review by sonologico — APPROVED** [review]
Looks good to me. Thanks for picking this up.

Don't worry about my PR. I'll sort it out. Running ocamlformat on my changes should make the conflicts manageable.

If there's another open PR that you think we should wait on, let's prioritize that. Then we can get this one out of the way and not have to think about formatting anymore.

**Pull request merged**